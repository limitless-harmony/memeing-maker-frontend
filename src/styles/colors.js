import Color from 'color';

export const black = '#000';
export const darkGrey = '#0F0F0F';
export const dark = '#4A4A4A';
export const white = '#FFF';
export const pink = '#EA2887';
export const buttonBorder = '#979797';
export const inputFill = '#F4F4F4';
export const green = '#01BF8D';

export const darken = (color, value) => {
  return Color(color)
    .darken(value)
    .hex();
};
