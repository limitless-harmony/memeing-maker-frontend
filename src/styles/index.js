export const baseSize = 16;

export const calculateRem = px => `${parseInt(px, 10) / baseSize}rem`;

export const mobileWidth = 420;

export const pxToVw = (px, width) => `${(100 * px) / width}vw`;

export const setFontSize = (text, small) => {
  const lowerCharacter = 24;
  const upperCharacter = 70;
  const lowerFontSize = small ? 8 : 20;
  const upperFontSize = small ? 14 : 28;
  const fontSizeRange = upperFontSize - lowerFontSize;
  const characterRange = upperCharacter - lowerCharacter;
  let fontSize =
    text.length > lowerCharacter
      ? upperFontSize -
        (fontSizeRange / characterRange) * (text.length - lowerCharacter)
      : upperFontSize;
  fontSize = fontSize > lowerFontSize ? fontSize : lowerFontSize;
  return fontSize;
};

export const setTextAreaHeight = (text, size) => {
  const lineCharacters = 24;
  const minHeight = 45;
  const maxHeight = size / 6;

  const { length } = text;
  const calculatedHeight = minHeight + (length - lineCharacters) * 2;
  const height =
    length <= lineCharacters
      ? minHeight
      : calculatedHeight < maxHeight
      ? calculatedHeight
      : maxHeight;
  return height;
};
