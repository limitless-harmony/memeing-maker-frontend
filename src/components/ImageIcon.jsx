import React from 'react';
import styled from 'styled-components';

import { calculateRem } from 'styles';

const ImageIcon = ({ height = 40, width = 40, src }) => {
  return <StyledIcon height={height} width={width} src={src} />;
};

const StyledIcon = styled.img`
  display: block;
  height: ${({ height }) => `${calculateRem(height)}`};
  width: ${({ width }) => `${calculateRem(width)}`};
`;

export default ImageIcon;
