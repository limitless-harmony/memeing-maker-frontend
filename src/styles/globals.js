import styledNormalize from 'styled-normalize';
import { createGlobalStyle } from 'styled-components';
import { baseSize, calculateRem } from '.';
import { white, black } from './colors';

const globalStyles = createGlobalStyle`
  ${styledNormalize}

  html, body {
    height: 100%;
    margin: 0;
  }

  body {
    background-color: ${white};
    color: ${black};
    font-size: ${baseSize}px;
    position:relative;
  }
  * {
    box-sizing: border-box;
  }
  #root {
    display: flex;
    min-height: 100%;
  }
  
  h2 {
    font-size: ${calculateRem(50)};
  }
  p {
    font-size: ${calculateRem(18)};
    line-height: 1.5;
  }
`;

export default globalStyles;
