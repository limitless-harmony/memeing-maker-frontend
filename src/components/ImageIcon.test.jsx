import React from 'react';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import ImageIcon from 'components/ImageIcon';
import { calculateRem } from 'styles';

describe('Testing the ImageIcon component', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<ImageIcon src="www.me.memememem" />);
    expect(wrapper.find('img'));
    expect(wrapper).toMatchSnapshot();
  });
  it('It changes the width and height', () => {
    const tree = renderer.create(<ImageIcon height={50} width={50} />).toJSON();
    expect(tree).toHaveStyleRule('height', `${calculateRem(50)}`);
    expect(tree).toHaveStyleRule('width', `${calculateRem(50)}`);
  });
});
