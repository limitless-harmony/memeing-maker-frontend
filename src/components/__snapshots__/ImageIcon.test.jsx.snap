// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the ImageIcon component renders without crashing 1`] = `
<styled.img
  height={40}
  src="www.me.memememem"
  width={40}
/>
`;
